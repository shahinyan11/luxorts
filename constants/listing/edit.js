import { compose, omit, values } from 'ramda';

import ROUTES from 'constants/routes';

// eslint-disable-next-line import/prefer-default-export
export const EDIT_LISTING_TABS = compose(
  values,
  omit(['KEY', 'PATH', 'META']),
)(ROUTES.DASHBOARD.LISTINGS);
