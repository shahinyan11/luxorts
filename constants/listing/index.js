import { v4 as uuid } from 'uuid';

import ROUTES from 'constants/routes';

export const LISTING_STEP_STATUS = {
  INITIAL: 'initial',
  ACCOMODATION: 'accomodation',
  LOCATION: 'location',
  LOCATION_MAP: 'location_map',
  AMENITIES: 'amenities',
  CAPABILITIES: 'capabilities',
  SERVICES: 'services',
  PHOTOS: 'photos',
  DESCRIPTION: 'description',
  HOUSE_RULES: 'house_rules',
  PREFERENCES: 'preferences',
  AVAILABILITY: 'availability',
  PRICING: 'pricing',
  FINISHED: 'finished',
};

export const LISTING_PATH_BY_STATUS = {
  [LISTING_STEP_STATUS.INITIAL]: ROUTES.ADD_NEW_LISTING.INDEX.PATH,
  [LISTING_STEP_STATUS.ACCOMODATION]: ROUTES.ADD_NEW_LISTING.ACCOMMODATION.PATH,
  [LISTING_STEP_STATUS.LOCATION]: ROUTES.ADD_NEW_LISTING.LOCATION.PATH,
  [LISTING_STEP_STATUS.LOCATION_MAP]: ROUTES.ADD_NEW_LISTING.LOCATION_MAP.PATH,
  [LISTING_STEP_STATUS.AMENITIES]: ROUTES.ADD_NEW_LISTING.AMENITIES.PATH,
  [LISTING_STEP_STATUS.CAPABILITIES]: ROUTES.ADD_NEW_LISTING.CAPABILITIES.PATH,
  [LISTING_STEP_STATUS.SERVICES]: ROUTES.ADD_NEW_LISTING.SERVICES.PATH,
  [LISTING_STEP_STATUS.PHOTOS]: ROUTES.ADD_NEW_LISTING.PHOTOS.PATH,
  [LISTING_STEP_STATUS.DESCRIPTION]: ROUTES.ADD_NEW_LISTING.DESCRIPTION.PATH,
  [LISTING_STEP_STATUS.HOUSE_RULES]: ROUTES.ADD_NEW_LISTING.HOUSE_RULES.PATH,
  [LISTING_STEP_STATUS.PREFERENCES]: ROUTES.ADD_NEW_LISTING.RESERVATION_PREFERENCES.PATH,
  [LISTING_STEP_STATUS.AVAILABILITY]: ROUTES.ADD_NEW_LISTING.AVAILABILITY.PATH,
  [LISTING_STEP_STATUS.PRICING]: ROUTES.ADD_NEW_LISTING.PRICING_AND_DISCOUNTS.PATH,
  [LISTING_STEP_STATUS.FINISHED]: ROUTES.ADD_NEW_LISTING.PUBLISH.PATH,
};

export const LISTING_PROGRESS_BY_STATUS = {
  [LISTING_STEP_STATUS.INITIAL]: 14,
  [LISTING_STEP_STATUS.ACCOMODATION]: 20,
  [LISTING_STEP_STATUS.LOCATION]: 26,
  [LISTING_STEP_STATUS.LOCATION_MAP]: 32,
  [LISTING_STEP_STATUS.AMENITIES]: 38,
  [LISTING_STEP_STATUS.CAPABILITIES]: 44,
  [LISTING_STEP_STATUS.SERVICES]: 50,
  [LISTING_STEP_STATUS.PHOTOS]: 56,
  [LISTING_STEP_STATUS.DESCRIPTION]: 62,
  [LISTING_STEP_STATUS.HOUSE_RULES]: 68,
  [LISTING_STEP_STATUS.PREFERENCES]: 74,
  [LISTING_STEP_STATUS.AVAILABILITY]: 80,
  [LISTING_STEP_STATUS.PRICING]: 86,
  [LISTING_STEP_STATUS.FINISHED]: 100,
};

export const LISTING_PLACE_TYPE = {
  ENTIRE_ROOM: 'entire_room',
  PRIVATE_PLACE: 'private_place',
  SHARED_ROOM: 'shared_room',
};

export const LISTING_PLACE_OPTIONS = [
  { value: LISTING_PLACE_TYPE.ENTIRE_ROOM, label: { id: 'shared.entirePlace' } },
  { value: LISTING_PLACE_TYPE.PRIVATE_PLACE, label: { id: 'shared.privateRoom' } },
  { value: LISTING_PLACE_TYPE.SHARED_ROOM, label: { id: 'shared.sharedRoom' } },
];

export const ACCOMMODATION_VALIDATION = {
  BASE: {
    MIN: 1,
    MAX: 16,
  },
  ARRANGEMENT: {
    MIN: 0,
    MAX: 16,
  },
};

export const BED_TYPE = {
  QUEEN: 'queen',
  SINGLE: 'single',
  DOUBLE: 'double',
  SOFA_BED: 'sofa_bed',
};

export const BED_NAME_BY_TYPE = {
  [BED_TYPE.DOUBLE]: { id: 'shared.double' },
  [BED_TYPE.QUEEN]: { id: 'shared.queen' },
  [BED_TYPE.SINGLE]: { id: 'shared.single' },
  [BED_TYPE.SOFA_BED]: { id: 'shared.sofaBed' },
};

export const LISTING_DEFAULT_BEDS = [
  {
    uuid: uuid(),
    kind: BED_TYPE.DOUBLE,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.QUEEN,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.SINGLE,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.SOFA_BED,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
];

export const DEFAULT_BEDROOM_CONFIG = {
  defaultBeds: LISTING_DEFAULT_BEDS,
  customBeds: [],
  bedsForDeleting: [],
};

export const INITIAL_BEDROOM = {
  uuid: uuid(),
  ...DEFAULT_BEDROOM_CONFIG,
};

export const BATH_TYPE = {
  BATHTUB: 'bathtub',
  SHOWER: 'shower',
  JETTED_TUB: 'jetted_tub',
  DOUBLE_VANITY: 'double_vanity',
};

export const BATH_NAME_BY_TYPE = {
  [BATH_TYPE.BATHTUB]: { id: 'shared.bathtub' },
  [BATH_TYPE.SHOWER]: { id: 'shared.shower' },
  [BATH_TYPE.JETTED_TUB]: { id: 'shared.jettedTub' },
  [BATH_TYPE.DOUBLE_VANITY]: { id: 'shared.doubleVanity' },
};

export const LISTING_DEFAULT_BATHS = [
  {
    uuid: uuid(),
    kind: BATH_TYPE.BATHTUB,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.SHOWER,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.JETTED_TUB,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.DOUBLE_VANITY,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
];

export const DEFAULT_BATHROOM_CONFIG = {
  defaultBaths: LISTING_DEFAULT_BATHS,
  customBaths: [],
  bathsForDeleting: [],
};

export const INITIAL_BATHROOM = {
  uuid: uuid(),
  ...DEFAULT_BATHROOM_CONFIG,
};

export const LISTING_STATUS = {
  IN_DRAFT: 'in_draft',
  ON_MODERATION: 'on_moderation',
  UNPUBLISHED: 'unpublished',
  PUBLISHED: 'published',
  BLOCKED: 'blocked',
  DELETED: 'deleted',
};

export const TIME_PERIOD_KIND = {
  AVAILABLE: 'available',
  BLOCKED: 'blocked',
  BOOKING: 'booking',
  PREPARATION: 'preparation',
};

export const TIME_PERIOD_DESCRIPTION_BY_KIND = {
  [TIME_PERIOD_KIND.BLOCKED]: { id: 'shared.blocked' },
  [TIME_PERIOD_KIND.PREPARATION]: { id: 'shared.preparationTime' },
  [TIME_PERIOD_KIND.BOOKING]: { id: 'shared.booked' },
};

export const TIME_PERIOD_ICON_BY_KIND = {
  [TIME_PERIOD_KIND.BLOCKED]: 'clock',
  [TIME_PERIOD_KIND.PREPARATION]: 'clock',
  [TIME_PERIOD_KIND.BOOKING]: 'clock',
};

export const LISTING_IMAGE_PLACEHOLDER = '/images/housing-preview-01.png';
