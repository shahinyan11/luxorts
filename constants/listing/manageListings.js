import { SORT_DIRECTIONS } from '../index';
import { LISTING_STATUS } from './index';

export const LISTING_TAG_BY_STATUS = {
  [LISTING_STATUS.IN_DRAFT]: {
    TITLE: { id: 'shared.draft' },
    CLASSNAME: 'tag--draft',
  },
  [LISTING_STATUS.ON_MODERATION]: {
    TITLE: { id: 'shared.onModeration' },
    CLASSNAME: 'tag--warning',
  },
  [LISTING_STATUS.UNPUBLISHED]: { TITLE: { id: 'shared.unpublished' }, CLASSNAME: 'tag--cancel' },
  [LISTING_STATUS.PUBLISHED]: {
    TITLE: { id: 'shared.published' },
    CLASSNAME: 'tag--success',
  },
  [LISTING_STATUS.BLOCKED]: { TITLE: { id: 'shared.blocked' }, CLASSNAME: 'tag--error' },
  [LISTING_STATUS.DELETED]: { TITLE: { id: 'shared.removed' }, CLASSNAME: 'tag--error' },
};

export const LISTING_PAGINATION = { number: 1, size: 8 };

export const LISTING_SORT_KEY = {
  BEDROOMS_AMOUNT: 'bedrooms_amount',
  BEDS_AMOUNT: 'beds_amount',
  BATHROOMS_AMOUNT: 'bathrooms_amount',
  GUESTS_NUMBER: 'guests_number',
  LOCATION: 'location',
  CREATED_AT: 'created_at',
  UPDATED_AT: 'updated_at',
  STATUS: 'status',
};

export const LISTING_SORT = {
  sortKey: LISTING_SORT_KEY.UPDATED_AT,
  direction: SORT_DIRECTIONS.DESCEND,
};

export const LISTING_TABLE_COLUMNS_KEY = {
  LISTING: 'listing',
  HIDDEN_TITLE: 'hiddenTitle',
  GUESTS: 'guests',
  BEDROOMS: 'bedrooms',
  BEDS: 'beds',
  BATHS: 'baths',
  LOCATION: 'location',
  DATE_CREATED: 'dateCreated',
  LAST_UPDATED: 'lastUpdated',
  STATUS: 'status',
  SETTINGS: 'settings',
};

export const LISTING_TABLE_SETTINGS = [
  {
    key: LISTING_TABLE_COLUMNS_KEY.LISTING,
    label: { id: 'shared.listing' },
    props: { fixed: 'left', width: 364 },
    hiddenInSettings: true,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.HIDDEN_TITLE,
    label: { id: 'shared.hiddenTitle' },
    props: { width: 280 },
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.GUESTS,
    label: { id: 'shared.guests' },
    props: { width: 100 },
    sortKey: LISTING_SORT_KEY.GUESTS_NUMBER,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.BEDROOMS,
    label: { id: 'shared.bedrooms' },
    width: 120,
    sortKey: LISTING_SORT_KEY.BEDROOMS_AMOUNT,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.BEDS,
    label: { id: 'shared.beds' },
    props: { width: 89 },
    sortKey: LISTING_SORT_KEY.BEDS_AMOUNT,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.BATHS,
    label: { id: 'shared.baths' },
    props: { width: 94 },
    sortKey: LISTING_SORT_KEY.BATHROOMS_AMOUNT,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.LOCATION,
    label: { id: 'shared.location' },
    props: { width: 240 },
    sortKey: LISTING_SORT_KEY.LOCATION,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.DATE_CREATED,
    label: { id: 'shared.dateCreated' },
    props: { width: 137 },
    sortKey: LISTING_SORT_KEY.CREATED_AT,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.LAST_UPDATED,
    label: { id: 'shared.lastUpdated' },
    props: { width: 138 },
    sortKey: LISTING_SORT_KEY.UPDATED_AT,
  },
  {
    key: LISTING_TABLE_COLUMNS_KEY.STATUS,
    label: { id: 'shared.status' },
    props: { width: 134 },
    sortKey: LISTING_SORT_KEY.STATUS,
  },
];

export const LISTING_TABLE_DEFAULT_COLUMNS_SETTINGS = [
  LISTING_TABLE_COLUMNS_KEY.LISTING,
  LISTING_TABLE_COLUMNS_KEY.HIDDEN_TITLE,
  LISTING_TABLE_COLUMNS_KEY.GUESTS,
  LISTING_TABLE_COLUMNS_KEY.BEDROOMS,
  LISTING_TABLE_COLUMNS_KEY.BEDS,
  LISTING_TABLE_COLUMNS_KEY.BATHS,
  LISTING_TABLE_COLUMNS_KEY.LOCATION,
  LISTING_TABLE_COLUMNS_KEY.DATE_CREATED,
  LISTING_TABLE_COLUMNS_KEY.LAST_UPDATED,
  LISTING_TABLE_COLUMNS_KEY.STATUS,
];

export const LISTING_FILTER_KEY = {
  STATUS_IN: 'status-in',
  BEDS_AMOUNT_EQ: 'beds_amount-eq',
  GUESTS_NUMBER_EQ: 'guests_number-eq',
  BATHROOMS_AMOUNT_EQ: 'bathrooms_amount-eq',
  BEDROOMS_AMOUNT_EQ: 'bedrooms_amount-eq',
  AMENITIES_IN: 'amenities-in',
};

export const ACCOMMODATION_LISTING_FILTER_KEYS = [
  LISTING_FILTER_KEY.BEDS_AMOUNT_EQ,
  LISTING_FILTER_KEY.GUESTS_NUMBER_EQ,
  LISTING_FILTER_KEY.BATHROOMS_AMOUNT_EQ,
  LISTING_FILTER_KEY.BEDROOMS_AMOUNT_EQ,
];

export const LISTING_FILTERS = {
  [LISTING_FILTER_KEY.STATUS_IN]: [],
  [LISTING_FILTER_KEY.BEDS_AMOUNT_EQ]: 0,
  [LISTING_FILTER_KEY.GUESTS_NUMBER_EQ]: 0,
  [LISTING_FILTER_KEY.BATHROOMS_AMOUNT_EQ]: 0,
  [LISTING_FILTER_KEY.BEDROOMS_AMOUNT_EQ]: 0,
  [LISTING_FILTER_KEY.AMENITIES_IN]: [],
};

export const ACCOMMODATION_VALIDATION = {
  MIN: 0,
  MAX: 16,
};

export const LISTING_STATUS_OPTIONS = [
  {
    label: { id: 'shared.published' },
    value: LISTING_STATUS.PUBLISHED,
  },
  {
    label: { id: 'shared.unpublished' },
    value: LISTING_STATUS.UNPUBLISHED,
  },
  {
    label: { id: 'shared.onModeration' },
    value: LISTING_STATUS.ON_MODERATION,
  },
  {
    label: { id: 'shared.draft' },
    value: LISTING_STATUS.IN_DRAFT,
  },
  {
    label: { id: 'shared.blocked' },
    value: LISTING_STATUS.BLOCKED,
  },
];

export const SUCCESS_MESSAGE_BY_STATUS = {
  [LISTING_STATUS.UNPUBLISHED]: { description: { id: 'listing.status.modal.unpublish.success' } },
  [LISTING_STATUS.PUBLISHED]: { description: { id: 'listing.status.modal.publish.success' } },
  [LISTING_STATUS.DELETED]: { description: { id: 'listing.status.modal.remove.success' } },
};

export const SUCCESS_MODAL_BY_STATUS = {
  [LISTING_STATUS.ON_MODERATION]: { modalType: 'PUBLISH_LISTING_MODAL' },
};
