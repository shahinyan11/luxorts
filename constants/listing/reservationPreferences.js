export const LISTING_PRIOR_NOTIFIED_TYPE = {
  SAME_DAY: 'same_day',
  ONE_DAY: '1_day',
  TWO_DAYS: '2_days',
  THREE_DAYS: '3_days',
  SEVEN_DAYS: '7_days',
};

export const LISTING_PRIOR_NOTIFIED_OPTIONS = [
  { value: LISTING_PRIOR_NOTIFIED_TYPE.SAME_DAY, label: { id: 'shared.sameDay' } },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.ONE_DAY,
    label: { id: 'shared.dayNumber', values: { number: 1 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.TWO_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 2 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.THREE_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 3 } },
  },
  {
    value: LISTING_PRIOR_NOTIFIED_TYPE.SEVEN_DAYS,
    label: { id: 'shared.dayNumberPlural', values: { number: 7 } },
  },
];

export const LISTING_IN_ADVANCE_BOOKING_TYPE = {
  ANY_TIME: 'any_time',
  THREE_MONTHS: '3_months_in_advance',
  SIX_MONTHS: '6_months_in_advance',
  NINE_MONTHS: '9_months_in_advance',
  ONE_YEAR: '1_year',
  UNAVAILABLE: 'unavailable',
};

export const LISTING_IN_ADVANCE_BOOKING_OPTIONS = [
  { value: LISTING_IN_ADVANCE_BOOKING_TYPE.ANY_TIME, label: { id: 'shared.anyTime' } },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.THREE_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 3 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.SIX_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 6 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.NINE_MONTHS,
    label: { id: 'shared.numberMonthsInAdvance', values: { number: 9 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.ONE_YEAR,
    label: { id: 'shared.yearNumber', values: { number: 1 } },
  },
  {
    value: LISTING_IN_ADVANCE_BOOKING_TYPE.UNAVAILABLE,
    label: { id: 'shared.datesUnavailableByDefault' },
  },
];

export const LISTING_RESERVATION_PREFERENCES_VALIDATION = {
  GUESTS_STAY_TIME: {
    MIN: 1,
    MAX: 30,
  },
  PREPARATION_TIME: 0,
};

export const LISTING_PREPARATION_TIME_OPTIONS = [
  {
    value: 0,
    label: { id: 'shared.none' },
  },
  {
    value: 1,
    label: { id: 'shared.blockNumberNightBeforeAndAfterEachReservation', values: { number: 1 } },
  },
  {
    value: 2,
    label: { id: 'shared.blockNumberNightBeforeAndAfterEachReservation', values: { number: 2 } },
  },
];

export const LISTING_APPROVAL_POLICY_TYPE = {
  INSTANT_BOOK: 'instant_book',
  TWENTY_FOUR_HOURS_REVIEW: '24_hours_review',
};

export const LISTING_APPROVAL_POLICY_OPTIONS = [
  {
    value: LISTING_APPROVAL_POLICY_TYPE.INSTANT_BOOK,
    label: { id: 'shared.instantBook' },
    description: { id: 'shared.allowsBookingImmediately' },
  },
  {
    value: LISTING_APPROVAL_POLICY_TYPE.TWENTY_FOUR_HOURS_REVIEW,
    label: { id: 'shared.24HourReview' },
    description: { id: 'shared.allows24Hours' },
  },
];

export const LISTING_CANCELLATION_POLICY_TYPE = {
  FLEXIBLE: 'flexible',
  MODERATE: 'moderate',
  STRICT: 'strict',
};

export const LISTING_CANCELLATION_POLICY_OPTIONS = [
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.FLEXIBLE,
    label: { id: 'shared.flexible' },
    description: { id: 'shared.fullRefundNumberDay', values: { number: 1 } },
  },
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.MODERATE,
    label: { id: 'shared.moderate' },
    description: { id: 'shared.fullRefundNumberDay', values: { number: 5 } },
  },
  {
    value: LISTING_CANCELLATION_POLICY_TYPE.STRICT,
    label: { id: 'shared.strict' },
    description: { id: 'shared.percentsRefundUpUntil', values: { percents: '50%' } },
  },
];
