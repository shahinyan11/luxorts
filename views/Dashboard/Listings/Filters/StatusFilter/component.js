import { Dropdown } from 'antd';
import PropTypes from 'prop-types';
import { FormikProvider } from 'formik';
import classNames from 'classnames';

import { LISTING_FILTER_KEY, LISTING_STATUS_OPTIONS } from 'constants/listing/manageListings';

import SvgIcon from 'views/shared/SvgIcon';
import CheckboxGroupField from 'views/shared/form/CheckboxGroupField';
import GradientButton from 'views/shared/GradientButton';

import useContainer from './hook';

const StatusFilter = ({ filters, onFilter }) => {
  const { visible, formik, onVisibilityChangeHandler, onApply, onClear } = useContainer({
    onFilter,
    filters,
  });
  const appliedFilters = filters[LISTING_FILTER_KEY.STATUS_IN].length;
  const hasFilters = appliedFilters > 0;

  return (
    <Dropdown
      trigger="click"
      visible={visible}
      onVisibleChange={onVisibilityChangeHandler}
      placement="bottomLeft"
      className="table-head__dropdown-payment"
      overlay={
        <div className="table-head__dropdown-content">
          <FormikProvider value={formik}>
            <CheckboxGroupField
              className="d-flex flex-column mb-0 table-head__checkboxes"
              name={LISTING_FILTER_KEY.STATUS_IN}
              items={LISTING_STATUS_OPTIONS}
            />
          </FormikProvider>
          <div className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end">
            {formik.values[LISTING_FILTER_KEY.STATUS_IN].length > 0 && (
              <GradientButton
                onClick={onClear}
                className="main-btn--medium mr-16"
                variant="tertiary"
                text={{ id: 'shared.clear' }}
              />
            )}
            <GradientButton
              onClick={onApply}
              className="main-btn--medium min-width-120"
              text={{ id: 'shared.apply' }}
            />
          </div>
        </div>
      }
    >
      <GradientButton
        className={classNames('main-btn--medium', hasFilters && 'main-btn--filter-active')}
        variant="filter"
        text={
          hasFilters
            ? {
                id: 'shared.statusCount',
                values: { count: <span className="main-btn__amount">{appliedFilters}</span> },
              }
            : { id: 'shared.status' }
        }
        addonAfter={<SvgIcon icon="arrow-down" className="icon-right" />}
      />
    </Dropdown>
  );
};

StatusFilter.propTypes = {
  filters: PropTypes.shape().isRequired,
  onFilter: PropTypes.func.isRequired,
};

export default StatusFilter;
