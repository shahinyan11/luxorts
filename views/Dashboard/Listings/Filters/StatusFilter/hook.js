import { useCallback } from 'react';
import { pick } from 'ramda';
import { useFormik } from 'formik';

import { LISTING_FILTER_KEY, LISTING_FILTERS } from 'constants/listing/manageListings';

import useImmutableCallback from 'hooks/useImmutableCallback';
import useAntdDropdownVisible from 'hooks/useAntdDropdownVisible';

function useContainer({ filters, onFilter }) {
  const [visible, onVisibilityChangeHandler, setVisible] = useAntdDropdownVisible();

  /**
   * Formik initialization
   */
  const formik = useFormik({
    enableReinitialize: true,
    initialValues: {
      ...filters,
    },
  });

  /**
   * Handle apply button click
   * @type {(function(): void)|*}
   */
  const onApply = useCallback(() => {
    onFilter(formik.values);
    setVisible(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [formik.values]);

  /**
   * Handle clear button click
   */
  const onClear = useImmutableCallback(() => {
    onFilter(pick([LISTING_FILTER_KEY.STATUS_IN], LISTING_FILTERS));
    formik.resetForm();
  });

  return {
    visible,
    formik,
    onVisibilityChangeHandler,
    onApply,
    onClear,
    setVisible,
  };
}

export default useContainer;
