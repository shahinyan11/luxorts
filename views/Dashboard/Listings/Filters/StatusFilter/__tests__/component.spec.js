import { shallow } from 'enzyme';

import { LISTING_FILTER_KEY } from 'constants/listing/manageListings';

import useContainer from '../hook';
import StatusFilter from '../component';

const mockedHookData = {
  visible: true,
  formik: {
    values: {
      [LISTING_FILTER_KEY.STATUS_IN]: [],
    },
  },
  onVisibilityChangeHandler: jest.fn(),
  onApply: jest.fn(),
  onClear: jest.fn(),
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('StatusFilter component', () => {
  const props = {
    filters: {
      [LISTING_FILTER_KEY.STATUS_IN]: [],
    },
    onFilter: jest.fn(),
  };
  let component = shallow(<StatusFilter {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when formik.values[LISTING_FILTER_KEY.STATUS_IN].length more than 0', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      formik: {
        values: {
          [LISTING_FILTER_KEY.STATUS_IN]: [{}],
        },
      },
    });

    component = shallow(<StatusFilter {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when filters[LISTING_FILTER_KEY.STATUS_IN].length more than 0', () => {
    component.setProps({
      filters: {
        [LISTING_FILTER_KEY.STATUS_IN]: [{}],
      },
    });

    expect(component).toMatchSnapshot();
  });
});
