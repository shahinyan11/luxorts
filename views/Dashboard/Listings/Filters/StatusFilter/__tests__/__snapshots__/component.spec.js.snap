// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusFilter component matches snapshot 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <Context.Provider
        value={
          Object {
            "values": Object {
              "status-in": Array [],
            },
          }
        }
      >
        <CheckboxGroupField
          className="d-flex flex-column mb-0 table-head__checkboxes"
          items={
            Array [
              Object {
                "label": Object {
                  "id": "shared.published",
                },
                "value": "published",
              },
              Object {
                "label": Object {
                  "id": "shared.unpublished",
                },
                "value": "unpublished",
              },
              Object {
                "label": Object {
                  "id": "shared.onModeration",
                },
                "value": "on_moderation",
              },
              Object {
                "label": Object {
                  "id": "shared.draft",
                },
                "value": "in_draft",
              },
              Object {
                "label": Object {
                  "id": "shared.blocked",
                },
                "value": "blocked",
              },
            ]
          }
          name="status-in"
        />
      </Context.Provider>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium"
    text={
      Object {
        "id": "shared.status",
      }
    }
    variant="filter"
  />
</Dropdown>
`;

exports[`StatusFilter component matches snapshot when filters[LISTING_FILTER_KEY.STATUS_IN].length more than 0 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <Context.Provider
        value={
          Object {
            "values": Object {
              "status-in": Array [],
            },
          }
        }
      >
        <CheckboxGroupField
          className="d-flex flex-column mb-0 table-head__checkboxes"
          items={
            Array [
              Object {
                "label": Object {
                  "id": "shared.published",
                },
                "value": "published",
              },
              Object {
                "label": Object {
                  "id": "shared.unpublished",
                },
                "value": "unpublished",
              },
              Object {
                "label": Object {
                  "id": "shared.onModeration",
                },
                "value": "on_moderation",
              },
              Object {
                "label": Object {
                  "id": "shared.draft",
                },
                "value": "in_draft",
              },
              Object {
                "label": Object {
                  "id": "shared.blocked",
                },
                "value": "blocked",
              },
            ]
          }
          name="status-in"
        />
      </Context.Provider>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium main-btn--filter-active"
    text={
      Object {
        "id": "shared.statusCount",
        "values": Object {
          "count": <span
            className="main-btn__amount"
          >
            1
          </span>,
        },
      }
    }
    variant="filter"
  />
</Dropdown>
`;

exports[`StatusFilter component matches snapshot when formik.values[LISTING_FILTER_KEY.STATUS_IN].length more than 0 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <Context.Provider
        value={
          Object {
            "values": Object {
              "status-in": Array [
                Object {},
              ],
            },
          }
        }
      >
        <CheckboxGroupField
          className="d-flex flex-column mb-0 table-head__checkboxes"
          items={
            Array [
              Object {
                "label": Object {
                  "id": "shared.published",
                },
                "value": "published",
              },
              Object {
                "label": Object {
                  "id": "shared.unpublished",
                },
                "value": "unpublished",
              },
              Object {
                "label": Object {
                  "id": "shared.onModeration",
                },
                "value": "on_moderation",
              },
              Object {
                "label": Object {
                  "id": "shared.draft",
                },
                "value": "in_draft",
              },
              Object {
                "label": Object {
                  "id": "shared.blocked",
                },
                "value": "blocked",
              },
            ]
          }
          name="status-in"
        />
      </Context.Provider>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium mr-16"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.clear",
            }
          }
          variant="tertiary"
        />
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium"
    text={
      Object {
        "id": "shared.status",
      }
    }
    variant="filter"
  />
</Dropdown>
`;
